### ISSUE TITLE
Reproduce VGT-AR Pretraining (InternVL3)

### ISSUE THREAD

**Issue by DingShizhe** [hustvl/VGT#12]
I’m pretraining **VGT InternVL3 0.6B (448px)** using the official **pretraining** pipeline, but the generation quality is much lower than expected. Training appears stable and the loss decreases normally, but visual quality remains poor and improves only slightly over time.

I attach generated samples from **2k, 50k, and 100k iterations**, as well as the loss curves.

### samples from 2k iterations

<img width="1523" height="452" alt="Image" src="https://github.com/user-attachments/assets/3b891f62-934d-4614-8ced-e9d4d7ff8d6b" />

### samples from 50k iterations

<img width="1523" height="443" alt="Image" src="https://github.com/user-attachments/assets/57831dac-83f6-41db-8775-89e6042ff520" />

### samples from 100k iterations

<img width="1495" height="394" alt="Image" src="https://github.com/user-attachments/assets/9e787804-de3e-4687-8ca0-f60b91f2fc1f" />

### loss curves

<img width="1570" height="624" alt="Image" src="https://github.com/user-attachments/assets/f84dac48-905b-4117-90fb-9b726b6d8ca6" />

---

### Training Setup

* 100k iterations, 8 GPUs (DDP)
* Global batch size: 256
* LR: 3e-4 peak, cosine decay to 1e-4
* Warmup: 1k iters
* AdamW (0.9, 0.95), weight decay 0.05
* EMA start 10k (momentum 0.0002)
* REPA loss weight 0.5

Config: `configs/pretrain/vgt_internvl3_0_6B_448px_pretrain.py`

---

### Dataset

Mixed training data:

* megalith10m
* text2image2m
* imagenet1k_t2i_qwenvl_flux

---

### Question

Is this level of generation quality expected for this setup, or is there anything important I should check or adjust?

Thanks — happy to share more details if needed.

**Comment by gjhhust** [comment]
Thank you for your attention. The training seems to be normal, but the pre-training data of about 12M appears to be too little.，You need to add more data like [datasets](https://github.com/hustvl/VGT/blob/main/docs/DATASETS.md), The low-quality data from megalith10m accounts for an excessively large proportion in your current ratio.

**Comment by DingShizhe** [comment]
So if I understand correctly, I can combine these datasets—**cc12m, text2image2m, redcaps5m, laion6m, imagenet1k_t2i_qwenvl_flux**—for pretraining and get promising results, right?

Is **midjourney_captioned_23m** really necessary? I’d like to use it, but it’s just too big.

**Comment by gjhhust** [comment]
Sure! Please contact gjh_hust@foxmail.com
